app/oxalis/security/OpenIdConnectClient.scala

```diff
@@ -51,42 +55,54 @@ class OpenIdConnectClient @Inject()(rpc: RPC, conf: WkConf)(implicit ec: Executi
   fields described by https://www.rfc-editor.org/rfc/rfc6749#section-4.4.2
    */
-  def getAndValidateToken(redirectUrl: String, code: String): Fox[JsObject] =
+  def getAndValidateToken(redirectUrl: String, code: String): Fox[(JsObject, Option[JsObject])] =
     for {
       _ <- bool2Fox(conf.Features.openIdConnectEnabled) ?~> "oidc.disabled"
       _ <- bool2Fox(oidcConfig.isValid) ?~> "oidc.configuration.invalid"
       serverInfos <- discover
       tokenResponse <- rpc(serverInfos.token_endpoint)
+        .silentIf(!conf.SingleSignOn.OpenIdConnect.verboseLoggingEnabled)
         .withBasicAuthOpt(Some(oidcConfig.clientId), oidcConfig.clientSecret)
         .postFormParseJson[OpenIdConnectTokenResponse](
           Map(
             "grant_type" -> "authorization_code",
             "client_id" -> oidcConfig.clientId,
             "redirect_uri" -> redirectUrl,
-            "code" -> code
+            "code" -> code,
+            "scope" -> oidcConfig.scope
           ))
-      newToken <- validateOpenIdConnectTokenResponse(tokenResponse, serverInfos) ?~> "failed to parse JWT"
-    } yield newToken
+      _ = logDebug(
+        s"Fetched oidc token for scope '${oidcConfig.scope}', response scope: ${tokenResponse.scope}. Decoding...")
+      (accessToken, idToken) <- validateOpenIdConnectTokenResponse(tokenResponse, serverInfos) ?~> "failed to parse JWT"
+      _ = logDebug(s"Got id token $idToken and access token $accessToken.")
+    } yield (accessToken, idToken)
 
   /*
   Discover endpoints of the provider (https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig)
    */
   private def discover: Fox[OpenIdConnectProviderInfo] =
     for {
-      response: WSResponse <- rpc(oidcConfig.discoveryUrl).get
+      response: WSResponse <- rpc(oidcConfig.discoveryUrl)
+        .silentIf(!conf.SingleSignOn.OpenIdConnect.verboseLoggingEnabled)
+        .get
       serverInfo <- response.json.validate[OpenIdConnectProviderInfo](OpenIdConnectProviderInfo.format)
     } yield serverInfo
 
-  private def validateOpenIdConnectTokenResponse(tokenResponse: OpenIdConnectTokenResponse,
-                                                 serverInfos: OpenIdConnectProviderInfo): Fox[JsObject] =
+  private def validateOpenIdConnectTokenResponse(
+      tokenResponse: OpenIdConnectTokenResponse,
+      serverInfos: OpenIdConnectProviderInfo): Fox[(JsObject, Option[JsObject])] =
     for {
       publicKey <- fetchServerPublicKey(serverInfos)
-      decodedResponse <- JwtJson.decodeJson(tokenResponse.access_token, publicKey).toFox
-    } yield decodedResponse
+      decdoedAccessToken <- JwtJson.decodeJson(tokenResponse.access_token, publicKey).toFox
+      decodedIdToken: Option[JsObject] <- Fox.runOptional(tokenResponse.id_token)(itToken =>
+        JwtJson.decodeJson(itToken, publicKey).toFox)
+    } yield (decdoedAccessToken, decodedIdToken)
```

Comment:
```suggestion
      decodedAccessToken <- JwtJson.decodeJson(tokenResponse.access_token, publicKey).toFox
      decodedIdToken: Option[JsObject] <- Fox.runOptional(tokenResponse.id_token)(itToken =>
        JwtJson.decodeJson(itToken, publicKey).toFox)
    } yield (decodedAccessToken, decodedIdToken)
```

Reply:
Also what does "itToken" mean here?

Reply:
oops, that was of course supposed to be idToken